raiden/storage/sqlite.py

```diff
@@ -66,12 +65,27 @@ def __init__(self, database_path, serializer):
         # condition.
         self.write_lock = threading.Lock()
         self.serializer = serializer
+        self._upgrade_callback = None
+
+    def register_upgrade_callback(self, callback: Callable[[int, int], None]):
+        if not callable(callback):
+            raise TypeError("Callback is not callable")
+        self._upgrade_callback = callback
+
+    def maybe_upgrade(self):
+        db_version = self.get_version()
+        if RAIDEN_DB_VERSION <= db_version:
+            return
+
+        if self._upgrade_callback:
```

Comment:
My initial thought was that: if the storage detects a difference in versions, it triggers the upgrade to happen via the callback. Will re-work this.

Reply:
Calling `manager.run()` from the raiden service seems the simplest option, specially since the upgrade functions are hardcoded in the upgrademanager anyways, having a callback here is not making this more generic.